src/WebJobs.Script.WebHost/Standby/StandbyManager.cs

```diff
@@ -107,12 +122,30 @@ public async Task InitializeAsync()
                     File.WriteAllText(Path.Combine(functionPath, "run.csx"), content);
 
                     _logger.LogInformation($"StandbyMode placeholder function directory created");
+
+                    // start a background timer to identify when specialization happens
+                    // specialization usually happens via an http request (e.g. scale controller
+                    // ping) but this timer is started as well to handle cases where we
+                    // might not receive a request
+                    _specializationTimer = new Timer(OnSpecializationTimerTick, null, _specializationTimerInterval, _specializationTimerInterval);
                 }
                 finally
                 {
                     _semaphore.Release();
                 }
             }
         }
+
+        private void OnSpecializationTimerTick(object state)
+        {
+            if (!_environment.InStandbyMode)
```

Comment:
There are 3 env vars involved in specialization: this one, ContainerReady and ConfigurationReady. We used to check all these here: https://github.com/Azure/azure-functions-host/blob/v1.x/src/WebJobs.Script.WebHost/App_Start/WebHostResolver.cs#L139. Where does that logic exist now?